test/sai_test/config/tunnel_configer.py

```diff
@@ -0,0 +1,157 @@
+
+    """
+    tunnel_configer = TunnelConfiger(test_obj)
+
+    if is_create_tunnel:
+        tunnel = tunnel_configer.create_tunnel([1], [17,18])
+        tunnel.tun_ip = test_obj.servers[11][1].ipv4
+        tunnel_configer.create_tunnel_term(tunnel)
+        test_obj.dut.tunnel_list.append(tunnel)
+
+    """
+    tunnel_configer.create_tunnel()
+    tunnel_configer.create_tunnel_term()
+    """
+
+
+class TunnelConfiger(object):
+    """
+    Class use to make all the Lag configurations.
+    """
+
+    def __init__(self, test_obj: 'T0TestBase') -> None:
+        """
+        Init Tunnel configrer.
+
+        Args:
+            test_obj: the test object
+        """
+        self.test_obj = test_obj
+        self.client = test_obj.client
+
+    def create_tunnel(self, oports, uports):
+        """
+        Create tunnel.
+
+        Args:
+            oports:  oport indexs
+            uports:  oport indexs
+
+        Returns:
+            Tunnel: tunnel object
+        """
+        tunnel: Tunnel = Tunnel(uport_indexs=uports, 
+                                oport_indexs=oports, 
+                                tunnel_type = SAI_TUNNEL_TYPE_IPINIP,
+                                term_type = SAI_TUNNEL_TERM_TABLE_ENTRY_TYPE_P2P)
+        
+        # underlay configuration
+        self.uvrf = self.test_obj.dut.default_vrf
+
+        # overlay configuration
+        self.ovrf = self.test_obj.dut.default_vrf
+
+        tunnel.urif_lpb = sai_thrift_create_router_interface(
+            self.client,
+            type=SAI_ROUTER_INTERFACE_TYPE_LOOPBACK,
+            virtual_router_id=tunnel.uvrf)
+
+        tunnel.orif_lpb = sai_thrift_create_router_interface(
+            self.client,
+            type=SAI_ROUTER_INTERFACE_TYPE_LOOPBACK,
+            virtual_router_id=tunnel.ovrf)
+
+        # tunnel
+        tunnel.oid = sai_thrift_create_tunnel(
+            self.client,
+            type=tunnel.tunnel_type,
+            encap_src_ip=sai_ipaddress(tunnel.lpb_ip),
+            underlay_interface=tunnel.urif_lpb,
+            overlay_interface=tunnel.orif_lpb)
+        
+        return tunnel
+
+    def create_tunnel_term(self, tunnel):
+        """
+        Create tunnel term.
+
+        Args:
+            tunnel:  tunnel object
+            tun_ip:  tun_ip
+        """
+        tunnel_term = sai_thrift_create_tunnel_term_table_entry(
+            self.client,
+            tunnel_type=tunnel.tunnel_type,
+            vr_id=tunnel.uvrf,
+            action_tunnel_id=tunnel.oid,
+            type=tunnel.term_type,
+            dst_ip=sai_ipaddress(tunnel.lpb_ip),
+            src_ip=sai_ipaddress(tunnel.tun_ip))
+        tunnel.tunnel_terms.append(tunnel_term)
+
+    def create_tunnel_route_v4_v6(self, tunnel, vm_ip, vm_ipv6):
+        """
+        Create tunnel special route v4 and v6.
+
+        Args:
+            tunnel:  tunnel object
+            vm_ip: dst ipv4
+            vm_ipv6: dst ipv6
+        """
+        # tunnel nexthop for VM
```

Comment:
tunnel object extends from routable object, then tunnel object has nhop v4, v6 and ip v4, v6 attribute